File: src/state/examples/alerttoggle/AlertsOff.js
'use strict';

class AlertsOff {
  toggle(ctx) {
    ctx.service.register(ctx.user);
    const AlertsOn = require('./AlertsOn');
    ctx.setState(new AlertsOn());
    console.log(`[Toggle] ${ctx.user.name} -> ON`);
  }
  deliver(_ctx, _event) { /* ignorado mientras OFF */ }
  label() { return 'OFF'; }
}

module.exports = AlertsOff;

File: src/state/examples/alerttoggle/AlertsOff.java
package state.examples.alerttoggle;

import observer.examples.lostpetalerts.LostPetEvent;

public class AlertsOff implements AlertToggleState {

    @Override
    public void toggle(AlertContext ctx) {
        ctx.getService().register(ctx.getUser());
        ctx.setState(new AlertsOn());
        System.out.println("[Toggle] " + ctx.getUser().getName() + " -> ON");
    }

    @Override
    public void deliver(AlertContext ctx, LostPetEvent event) {
        // Ignored while OFF.
    }

    @Override public String label() { return "OFF"; }
}
